src/main/java/com/user_service/controllers/UserAuthenticationController.java

```diff
@@ -16,13 +16,13 @@ public class UserAuthenticationController {
     private UserAuthenticationService userAuthenticationService;
 
     @PostMapping(value = "/register")
-    public ResponseEntity<String> signUpUser(@RequestBody RegisterUserDto registerUserDto) {
-        SignUpResponseDto signUpResponseDto = userAuthenticationService.signUpUser(registerUserDto);
+    public ResponseEntity<String> signUpUser(@RequestBody SignUpUserDto signUpUserDto) {
+        SignUpResponseDto signUpResponseDto = userAuthenticationService.signUpUser(signUpUserDto);
 
         if (signUpResponseDto.getSignUpResponse() == null) {
```

Comment:
I would avoid such check in the controller layer. It leaks the implementation logic of Cognito here. The best would be to include in the dtos returned from the `userAuthenticationService` only information needed, to construct a response. It might be some `userId`, `token` ot something similar. Also, avoid nulls and use `Optional` instead.